ctxpropagation/starter/main.go

```diff
@@ -4,23 +4,29 @@ import (
 	"context"
 	"log"
 
-	"github.com/opentracing/opentracing-go"
 	"github.com/pborman/uuid"
+	"github.com/temporalio/samples-go/ctxpropagation"
 	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/contrib/opentracing"
+	"go.temporal.io/sdk/interceptor"
 	"go.temporal.io/sdk/workflow"
-
-	"github.com/temporalio/samples-go/ctxpropagation"
 )
 
 func main() {
 	// Set tracer which will be returned by opentracing.GlobalTracer().
 	closer := ctxpropagation.SetJaegerGlobalTracer()
 	defer func() { _ = closer.Close() }()
 
+	// Create interceptor
+	tracingInterceptor, err := opentracing.NewInterceptor(opentracing.TracerOptions{})
```

Comment:
One to keep in the back of your mind - a set of `MustNewInterceptor` functions might be useful to clients.

Reply:
True. In fact, you'll note internally this can never return an error right now and I just added error capability for future proofing if we add options needing more advanced validation. So it's a perfect candidate for panicking.

Issue opened at https://github.com/temporalio/sdk-go/issues/664